sklearn/utils/_array_api.py

```diff
@@ -108,6 +109,15 @@ def _check_array_api_dispatch(array_api_dispatch):
                 f"NumPy must be {min_numpy_version} or newer to dispatch array using"
                 " the API specification"
             )
+
+        scipy_version = parse_version(scipy.__version__)
+        min_scipy_version = "1.14.0"
+        if scipy_version < parse_version(min_scipy_version):
+            raise ImportError(
+                f"SciPy must be {min_scipy_version} or newer to dispatch array using"
+                " the API specification"
```

Comment:
It would be nice to add the actual scipy version found to the error message. I could try a Github suggestion but I am afraid I will mess it up because for example of black ...

In general I kind of think error message like "expected this, got that instead"

Reply:
Done. I adjusted the Numpy error message above as well.